game-core/src/main/java/games/strategy/triplea/attachments/AbstractPlayerRulesAttachment.java

```diff
@@ -328,4 +337,87 @@ public void resetNegateDominatingFirstRoundAttack() {
   public void validate(final GameData data) {
     validateNames(m_movementRestrictionTerritories);
   }
+
+  private static Map<String, Function<IAttachment, AttachmentProperty<?>>> getPopulatedAttachmentMap() {
```

Comment:
I think an enum would fit very well and could help organize the attachmentMap code more cleanly.

eg:

```
AttachmentMap.PLAYER_RULES.getAttachmentMap();
```

Then in enum you woudl consolidate this code and have a mega param, eg:

```
public enum AttachmentMap {
 PLAYER_LISTING( ImmutableMap.<String, Function<IAttachment, AttachmentProperty<?>>>builder()
    : 
     :);
   private final Map<String, Function<IAttachment, AttachmentProperty<?>>> map; 

    AttachmentMap(  Map<String, Function<IAttachment, AttachmentProperty<?>>> map) {
             this.map = map;
    }

}

```

Reply:
I'm not sure how exactly you are imagining this, a real example of one of the smaller attachment classes would be nice

Reply:
Here is a more concrete example: https://github.com/triplea-game/triplea/commit/69b06fe42e8012da0db6cd1fb99cc2a08938afb9

Reply:
Thanks, makes sense now.